realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -1553,6 +1553,10 @@ private void emitCreateOrUpdateUsingJsonObject(JavaWriter writer) throws IOExcep
         writer.emitEmptyLine();
     }
 
+    // FIXME: Since we need to check the PK in stream before create an object, this is now using copyToRealm instead of
```

Comment:
I was thinking something like we do byte code insertion a special constructor there, something like `Foo(RealmDefaultConstructor c)` if user doesn't have one there. With the construction, user can set the default values.